pkg/agent/route/route_windows.go

```diff
@@ -213,109 +236,121 @@ func (c *Client) DeleteRoutes(podCIDR *net.IPNet) error {
+		route.DestinationSubnet = curClusterIPCIDR
+		staleRoutes = []*util.Route{route}
+
+		// TODO: delete the code below in the future. The code below is used to delete stale route for the ClusterIP before
+		//   this commit. For Antrea which is not upgraded from the version before this commit, the code below is unnecessary.
+		for _, rt := range routes {
+			ones, _ := rt.DestinationSubnet.Mask.Size()
+			if ones == ipv4AddrLength && rt.DestinationSubnet.IP.Equal(svcIP) {
+				staleRoutes = append(staleRoutes, rt)
+			}
 		}
-		// Remove the existing route if gateway or metric is not as expected.
-		if err := util.RemoveNetRoute(existingRoute); err != nil {
-			return fmt.Errorf("failed to delete existing Service route entry, DestinationIP: %s, Gateway: %s, RouteMetric: %d, err: %v",
-				existingRoute.DestinationSubnet, existingRoute.GatewayAddress, existingRoute.RouteMetric, err)
+	} else {
+		// If current ClusterIP CIDR doesn't exist, which means that Antrea Agent has just started, and the route for
+		// ClusterIP generated in the last round should be deleted.
+		for _, rt := range routes {
+			// Skip the route whose destination CIDR only has one address.
+			ones, _ := rt.DestinationSubnet.Mask.Size()
+			if ones == ipv4AddrLength {
+				continue
+			}
+			// Find stale route.
+			if rt.DestinationSubnet.Contains(svcIP) && rt.GatewayAddress.Equal(gw) {
+				staleRoutes = append(staleRoutes, rt)
+				break
+			}
 		}
 	}
 
-	if err := util.ReplaceNetRoute(route); err != nil {
-		return err
+	// Remove stale routes.
+	for _, rt := range staleRoutes {
+		if err = util.RemoveNetRoute(rt); err != nil {
+			if strings.Contains(err.Error(), "No matching MSFT_NetRoute objects") {
+				klog.InfoS("Failed to delete stale ClusterIP route since the route doesn't exist", "route", route)
+			} else {
+				return fmt.Errorf("failed to delete route for ClusterIP %s: %w", svcIP.String(), err)
+			}
+		}
+		klog.V(4).InfoS("Uninstalled stale ClusterIP route successfully", "stale route", rt)
 	}
 
-	c.hostRoutes.Store(route.DestinationSubnet.String(), route)
-	klog.V(2).InfoS("Added Service route", "ServiceIP", route.DestinationSubnet, "GatewayIP", route.GatewayAddress)
 	return nil
 }
 
-func (c *Client) deleteServiceRoute(svcIP net.IP) error {
-	svcIPNet := util.NewIPNet(svcIP)
-	obj, found := c.hostRoutes.Load(svcIPNet.String())
-	if !found {
-		klog.V(2).InfoS("Service route does not exist", "DestinationIP", svcIP)
-		return nil
-	}
+func (c *Client) addVirtualNodePortDNATIPRoute(isIPv6 bool) error {
+	linkIndex := c.nodeConfig.GatewayConfig.LinkIndex
+	vIP := config.VirtualNodePortDNATIPv4
+	gw := config.VirtualServiceIPv4
 
-	rt := obj.(*util.Route)
-	if err := util.RemoveNetRoute(rt); err != nil {
-		return err
+	route := generateRoute(virtualNodePortDNATIPv4Net, gw, linkIndex, util.MetricHigh)
+	if err := util.ReplaceNetRoute(route); err != nil {
+		return fmt.Errorf("failed to install route for virtual Service IP %s: %w", vIP.String(), err)
```

Comment:
I think this is not for the "virtual service IP"

Reply:
Thanks, I'll rephrase the log.